up-l1/zenoh.adoc

```diff
@@ -86,93 +96,72 @@ Different message types **MUST** use different Zenoh API.
 
-It's a local UUri, so it starts with `upl`.
-Also, we leave the second part blank since it doesn't have UAuthority.
-Only ID in UEntity and UResource matters while serializing into link:../basics/uri.adoc#421-local-micro-uri[micro-form Uri].
-Then transform it into lowercase hexadecimal representation.
-The Zenoh key is `upl/0100162e04d20100`.
+[%autowidth]
+|===
+| Message Type | Zenoh Key
 
-* The remote UUri with UAuthority
+| Publish | up/src-authority/{}/src-entity/src-version/src-resource
+| Notification | up/src-authority/dst-authority/dst-entity/dst-version/dst-resource
+| Reqest | up/src-authority/dst-authority/dst-entity/dst-version/dst-resource
+| Response | Don't need it since relying on Zenoh queryable mechanism
+|===
 
-[source]
-----
-UUri {
-  authority: UAuthority {
-    name: UAuthName,
-    number: Id({01, 02, 03, 10, 11, 12})
-  },
-  entity: empty,
-  resource: empty,
-}
-----
+Note that since authority is case-insensitvie and allows some special characters defined in link:https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2[RFC3986, Section 3.2.2]. Zenoh needs to use `{}` to escape these characters and transform all the alphbets to lowercase.
 
-It's a remote UUri, so it starts with `upr`.
-The UAuthority is serialized into mirco form.
-Given how a UAuthority containing an ID is serialized into micro form as outlined in link:../basics/uri.adoc#422-remote-micro-uris[the spec],
-it is length + serialized UUri in lowercase hexadecimal representation, which is `06` + `0102030a0b0c`.
-The UEntity and UResource are all blank, so it is the special form UUri.
-It means that the third part is `+++**+++`.
-Therefore, the Zenoh key is `upr/060102030a0b0c/+++**+++`.
+[%autowidth]
+|===
+| Speical Characters in authority | Zenoh Key
 
-=== UPayload
+| empty | `{}`
+| `$` | `{dollar}`
+|===
 
-Although UPayload can be encoded into protobuf, it saves a lot of time without encoding.
-UPayload consists of three parts: length, format, data.
+Take some examples:
 
-* Length only works with shared memory, and we ignore it temporarily.
-* Format **SHOULD** be sent with Zenoh encoding. Zenoh supports `Encoding::WithSuffix`. The encoding here is `KnownEncoding::AppCustom` and the UPayloadFormat (int32) is transformed into the string.
+[%autowidth]
+|===
+| Use Case | Source UUri | Sink UUri | Zenoh Key
 
-[source, rust]
-----
-Encoding::WithSuffix(
-    KnownEncoding::AppCustom,
-    payload.format.value().to_string().into(),
-)
-----
+| Publish | up://192.168.1.100/1002/3/800A | - | up/192.168.1.100/{}/10002/3/800A
+| Notification | up://192.168.1.99/... | up://192.168.1.100/1002/3/800A | up/192.168.1.99/192.168.1.100/10002/3/800A
+| Request | up://MYHOST1/... | up://myhost$/1002/3/B | up/myhost1/myhost{dollar}/10002/3/B
+| uStreamer | up://+++*+++/... | up://192.168.1.100/FFFF/FF/FFFF | up/+++*+++/192.168.1.100/+++*+++/+++*+++/+++*+++
```

Comment:
Where do these `+` signs come from?

Reply:
Used to escape asterisk.
https://asciidoc-py.github.io/userguide.html#X77